app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepostoryImpl.kt

```diff
@@ -38,59 +38,68 @@ class AgendaRepositoryImpl @Inject constructor(
         return events + tasks // + reminders
     }
 
+    override suspend fun updateAgendaForDayFromRemote(dateTime: ZonedDateTime) {
+        CoroutineScope(Dispatchers.IO).launch {
```

Comment:
I think I commented on this before, but this scope shouldn't be here. You're inside a suspend function, so you can safely execute your I/O functions. Otherwise, you also won't have any way for a caller to know when this updating finished, since nothing will suspend here.